packages/taco/src/taco.ts

```diff
@@ -41,14 +76,23 @@ import { retrieveAndDecrypt } from './tdec';
  *
  * @throws {Error} If the active DKG Ritual cannot be retrieved an error is thrown.
  */
-export const encrypt = async (
-  provider: ethers.providers.Provider,
+export async function encrypt(
+  viemPublicClient: PublicClient,
```

Comment:
Let's stay consistent with the naming - also EIP4361AuthProvider uses `publicClient`. 

It's already a viem object so I don't think we need the `viem`-prefix. 
```suggestion
  publicClient: PublicClient,
```